test/test.dart

```diff
@@ -84,157 +86,192 @@ const kScreenWidth = 450.0;
+        Directory('./temp').createSync();
+        return './temp';
+      }
+      return null;
+    });
+    defaultBinaryMessenger.setMockMethodCallHandler(const MethodChannel('com.tekartik.sqflite'),
+        (MethodCall methodCall) async {
+      if (methodCall.method == 'getDatabasesPath') {
+        Directory('./temp').createSync();
+        return './temp';
+      }
+      if (methodCall.method == 'openDatabase') {
+        return 0;
+      }
+      if (methodCall.method == 'query') {
+        return {};
+      }
+      if (methodCall.method == 'execute') {
+        return null;
+      }
+      return null;
+    });
+    LicenseRegistry.reset();
+    LicenseRegistry.addLicense(() => Stream.value(const FakeLicenseEntry()));
+
+    configureFakes?.call();
+
+    // Set up regular classes
+    PlaybackControl.instance = PlaybackControl();
+    ContentControl.instance = ContentControl();
+
+    // Default initialization process from main.dart
+    await NFPrefs.initialize();
+    await SearchHistory.instance.clear();
+
+    await DeviceInfoControl.instance.init();
+    ThemeControl.instance.init();
+    ThemeControl.instance.initSystemUi();
+    await Permissions.instance.init();
+    final contentInitFuture = ContentControl.instance.init();
+    // Flush micro-tasks to allow all stream events to propagate to their listeners.
+    await pump(const Duration(seconds: 1));
+    await contentInitFuture;
+    // Called in the [App] widget
+    NFWidgets.init(
+      navigatorKey: AppRouter.instance.navigatorKey,
+      routeObservers: [routeObserver, homeRouteObserver],
+    );
   }
-
-  // Called in the [App] widget
-  NFWidgets.init(
-    navigatorKey: AppRouter.instance.navigatorKey,
-    routeObservers: [routeObserver, homeRouteObserver],
-  );
 }
 
 extension WidgetTesterExtension on WidgetTester {
-  /// A template for this:
-  ///  1. pumps the app
-  ///  2. runs the test from the [callback]
-  ///  3. stops and disposes the player
-  ///  4. optionally, runs [goldenCaptureCallback]. It would time out if was ran before player is disposed in [callback].
-  ///  5. un-pumps the screen
-  Future<void> runAppTest(AsyncCallback callback, {AsyncCallback? goldenCaptureCallback}) async {
+  /// Run a test that uses the App UI using the following procedure:
+  ///  1. Initializes the application, optionally running [initialization] when all fakes are set up
+  ///     before the [ContentControl] is initialized.
+  ///  2. Optionally, runs [postInitialization] after [ContentControl] is initialized.
+  ///  3. Pumps the app.
+  ///  4. Runs the test from the [callback].
+  ///  5. Optionally, runs [goldenCaptureCallback].
+  ///  6. Stops and disposes the player.
+  ///  7. Un-pumps the screen and flushes all micro-tasks and stream events.
+  Future<void> runAppTest(
+    AsyncCallback callback, {
+    AsyncCallback? goldenCaptureCallback,
+    VoidCallback? initialization,
+    VoidCallback? postInitialization,
```

Comment:
Should we align `runAppTestWithoutUi` with this one and pass this callback as well?